locust/dispatch.py

```diff
@@ -307,26 +327,58 @@ def _user_gen(self) -> Generator[str, None, None]:
         weighted round-robin algorithm, we'd get AAAAABAAAAAB which would make the distribution
         less accurate during ramp-up/down.
         """
-        # Normalize the weights so that the smallest weight will be equal to "target_min_weight".
-        # The value "2" was experimentally determined because it gave a better distribution especially
-        # when dealing with weights which are close to each others, e.g. 1.5, 2, 2.4, etc.
-        target_min_weight = 2
-        min_weight = min(u.weight for u in self._user_classes)
-        normalized_weights = [
-            (user_class.__name__, round(target_min_weight * user_class.weight / min_weight))
-            for user_class in self._user_classes
-        ]
-        gen = smooth(normalized_weights)
-        # Instead of calling `gen()` for each user, we cycle through a generator of fixed-length
-        # `generation_length_to_get_proper_distribution`. Doing so greatly improves performance because
-        # we only ever need to call `gen()` a relatively small number of times. The length of this generator
-        # is chosen as the sum of the normalized weights. So, for users A, B, C of weights 2, 5, 6, the length is
-        # 2 + 5 + 6 = 13 which would yield the distribution `CBACBCBCBCABC` that gets repeated over and over
-        # until the target user count is reached.
-        generation_length_to_get_proper_distribution = sum(
-            normalized_weight[1] for normalized_weight in normalized_weights
-        )
-        yield from itertools.cycle(gen() for _ in range(generation_length_to_get_proper_distribution))
+
+        def infinite_cycle_gen(users: List[Tuple[User, int]]) -> Generator[Optional[str], None, None]:
+            if not users:
+                return itertools.cycle([None])
+
+            # Normalize the weights so that the smallest weight will be equal to "target_min_weight".
+            # The value "2" was experimentally determined because it gave a better distribution especially
+            # when dealing with weights which are close to each others, e.g. 1.5, 2, 2.4, etc.
+            target_min_weight = 2
+
+            # 'Value' here means weight or fixed count
+            normalized_values = [
+                (
+                    user.__name__,
+                    round(target_min_weight * value / min([u[1] for u in users])),
+                )
+                for user, value in users
+            ]
+            generation_length_to_get_proper_distribution = sum(
+                normalized_val[1] for normalized_val in normalized_values
+            )
+            gen = smooth(normalized_values)
+            # Instead of calling `gen()` for each user, we cycle through a generator of fixed-length
```

Comment:
Just to be consistent with the vertical spacing above :) 
```suggestion
            gen = smooth(normalized_values)

            # Instead of calling `gen()` for each user, we cycle through a generator of fixed-length
```

Reply:
Done.